### PR TITLE
Fix #1691: Make asUnit(v) accept v === null.

### LINKED ISSUE #1691: asUnit(v) throws for v === null
It shouldn't. It should return `v`, like all the other `asXYZ` accept `null`.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki

File: test-suite/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala [hunk 1]
```diff
@@ -432,5 +432,11 @@ object RegressionTest extends JasmineTest {
       }
       expect(x).toEqual(5)
     }
+
+    it("null.asInstanceOf[Unit] should succeed - #1691") {
+      def getNull(): Any = null // s
```
Comment (nicolasstucki) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala]: What does this comment mean? `// s` Or is it just something that should not be there.

Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/compiler/RegressionTest.scala]: It should not be there ^^

**Comment by nicolasstucki** [comment]
LGTM

**Pull request merged**